src/dune_engine/sandbox.ml

```diff
@@ -83,11 +101,29 @@ let link_deps t ~mode ~deps =
             [ ("path", Path.to_dyn path) ]
       | Some p -> link path (Path.build (map_path t p)))
 
-let create ~mode ~deps ~rule_dir ~chdirs ~rule_digest ~expand_aliases =
+let snapshot t =
+  let rec walk dir acc =
+    match Path.Untracked.readdir_unsorted dir with
```

Comment:
We have this kind of directory traversal code in a few places already. It seems worth adding a CR-someday to unify all these traversals and remove code duplication.

Reply:
added

Reply:
Thanks!